internal/sink/sink.go

```diff
@@ -0,0 +1,544 @@
+package sink
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/signal"
+	"regexp"
+	"syscall"
+	"time"
+
+	"github.com/confluentinc/confluent-kafka-go/v2/kafka"
+
+	"github.com/openmeterio/openmeter/internal/dedupe"
+	"github.com/openmeterio/openmeter/internal/ingest/kafkaingest/serializer"
+)
+
+var namespaceTopicRegexp = regexp.MustCompile(`^om_([A-Za-z0-9]+(?:_[A-Za-z0-9]+)*)_events$`)
+var defaultDeadletterTopicTemplate = "om_%s_events_deadletter"
+
+type SinkMessage struct {
+	Namespace    string
+	KafkaMessage *kafka.Message
+	Serialized   *serializer.CloudEventsKafkaPayload
+	Error        *ProcessingError
+}
+
+type Sink struct {
+	consumer         *kafka.Consumer
+	producer         *kafka.Producer
+	config           *SinkConfig
+	running          bool
+	buffer           SinkBuffer
+	flushTimer       *time.Timer
+	namespaceStore   *NamespaceStore
+	namespaceRefetch *time.Timer
+}
+
+type SinkConfig struct {
+	Context                 context.Context
```

Comment:
This is generally an anti-pattern. Unless there is a very good reason, a context should be created for each request.